crypto.go

```diff
@@ -0,0 +1,59 @@
+package skynet
+
+import (
+	"encoding/binary"
+	"errors"
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/crypto/ed25519"
+)
+
+// encodeNumber converts the given number into a byte array.
+func encodeNumber(number uint64) []byte {
+	b := make([]byte, 8)
+	binary.LittleEndian.PutUint64(b, number)
+	return b
+}
+
+// encodeString converts the given string into a byte array.
+func encodeString(toEncode string) []byte {
```

Comment:
This function is encoding the length of the string, not the string itself. Is that intentional?